test/dsa_no_digest_size_test.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "testutil.h"
+
+#include <openssl/evp.h>
+#include <openssl/dsa.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+DSA *dsa;
+
+static int genkeys() {
+    dsa = DSA_new();
```

Comment:
Open curly of a function starts a line.

You should use "TEST_ptr(dsa = DSA_new())"
You don't need the TEST_error calls, TEST_ptr etc print source line/file.
So this becomes
```
if (!TEST_ptr(...)
    || !TEST_eq(DSA_generate...., 1)
    || !TEST_eq(DSA_generate_key... 1))
       return 0;
```
And similar for the other tests below.

Reply:
Also note that TEST_ptr appears to be missing on the 1.1.0 stable branch, so the backport in #6750 will start to diverge significantly here.